### PR TITLE
Update to new pageguard

### PR DESCRIPTION
**Pull request by zavies97**
Still needs testing in an env, and to then be added to all pages if it does work

### CODE REVIEW

File: app/config/question-bank.js [hunk 1]
```diff
@@ -144,6 +144,12 @@ const questionBank = {
           url: 'applicant',
           baseUrl: 'applicant',
           preValidationKeys: ['projectSubject'],
+          // preValidationObject: {
+          //    preValidationKey: ['projectSubject'],
+          //    preValidationAnswer: ['project-subject-A1'],
+          //    preValidationRule: 'AND',
+          //    preValidationUrls: ['project-subject']
+          // }
```
Comment (diya2214) [inline app/config/question-bank.js:152]: can we remove these comments please @zavies97

File: app/helpers/page-guard.js [hunk 2]
```diff
@@ -13,11 +14,93 @@ function guardPage (request, guardData, rule = null) {
...
+       key: guardData.preValidationKeys[i], 
+       values: (guardData.preValidationAnswer.filter((answer) => answer.startsWith(guardData.preValidationUrls[i]))),
+       url: guardData.preValidationUrls[i]
+      }
+      );
     }
+
+    console.log('should be a formatted list combining all answewrs and relevant values in one object', preValidationList)
+
+    // should format preValidations as below
```
Comment (diya2214) [inline app/helpers/page-guard.js:34]: do we need a console log here

File: app/helpers/page-guard.js [hunk 3]
```diff
@@ -13,11 +14,93 @@ function guardPage (request, guardData, rule = null) {
...
+    // [{
+    //   key: 'key name',
+    //   values: ['value 1', 'value 2'],
+    //   url: 'url'
+    // }]
+
```
Comment (diya2214) [inline app/helpers/page-guard.js:40]: delete comments please

File: app/helpers/page-guard.js [hunk 4]
```diff
@@ -13,11 +14,93 @@ function guardPage (request, guardData, rule = null) {
...
+          if (preValidationList[i].values.filter((answer) => getQuestionAnswer(preValidationList[i].url, answer) === getYarValue(request, preValidationList[i].key)).length > 0) {
+            return false
+          }
+        }
+        return true
+
+      case 'NOT':
+        // check if answer exists in list (if key and value pair contains needed answer)
+
+         for (let i = 0; i< preValidationList.length; i++) {
```
Comment (diya2214) [inline app/helpers/page-guard.js:65]: again can we use map @zavies97

File: app/helpers/page-guard.js [hunk 5]
```diff
@@ -13,11 +14,93 @@ function guardPage (request, guardData, rule = null) {
...
+
+      case 'SOLARAND':
+        // check for a type of journey and another requirement (solar + project repsonsibility etc)
+
+        // check yar value of journey
+        if (getYarValue(request, 'projectSubject') === getQuestionAnswer('projectSubject', 'project-subject-A1')) {
+          return true
+        }
+
+        for (let i = 0; i< preValidationList.length; i++) {
```
Comment (diya2214) [inline app/helpers/page-guard.js:81]: here as well map

File: app/helpers/page-guard.js [hunk 6]
```diff
@@ -13,11 +14,93 @@ function guardPage (request, guardData, rule = null) {
...
+        return true
+
+      case 'ROBOTAND':
+        // check for a type of journey and another requirement (solar + project repsonsibility etc)
+
+        if (getYarValue(request, 'projectSubject') === getQuestionAnswer('project-subject', 'project-subject-A2')) {
+          return true
+        }
+
+        for (let i = 0; i< preValidationList.length; i++) {
```
Comment (diya2214) [inline app/helpers/page-guard.js:95]: here too -> map

File: app/config/question-bank.js (hunk 1 above)
Comment (zavies97) [reply app/config/question-bank.js:152]: not yet as this is what will be added to question bank, and i havent actually added the pre validation yet as cant test it

File: app/helpers/page-guard.js [hunk 7]
```diff
@@ -13,11 +14,93 @@ function guardPage (request, guardData, rule = null) {
...
-      result = rule.condition === 'ANY' && !guardData.some(dependcyKey => getYarValue(request, dependcyKey) !== null)
-    } else {
-      result = guardData.filter(dependcyKey => getYarValue(request, dependcyKey) === null).length > 0
+    // filter list of answers with keys?
+
+    // make an object of question as key and answers for said question?
+
+    let preValidationList = []
+
+    for(let i=0; i<guardData.preValidationKeys.length; i++) {
```
[the remark this answers is not in the document]
Comment (zavies97) [reply app/helpers/page-guard.js:23]: maybe, havent been able to get this running in the first place tho :P

nor am i good at writing .map()

File: app/helpers/page-guard.js (hunk 2 above)
Comment (zavies97) [reply app/helpers/page-guard.js:34]: nope, its for my tesating when im able to run the app, to check what the key/value object looks like when its generated

File: app/helpers/page-guard.js (hunk 4 above)
Comment (zavies97) [reply app/helpers/page-guard.js:65]: maybe, havent tested yet tho and not good at writing .map() functions

File: app/helpers/page-guard.js (hunk 5 above)
Comment (zavies97) [reply app/helpers/page-guard.js:81]: same comment as others

File: app/helpers/page-guard.js (hunk 6 above)
Comment (zavies97) [reply app/helpers/page-guard.js:95]: same comment as others :)

File: app/helpers/page-guard.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (zavies97) [reply app/helpers/page-guard.js:23]: Update: i dont see a benefit to replacing the for as a .map

File: app/helpers/page-guard.js (hunk 4 above)
Comment (zavies97) [reply app/helpers/page-guard.js:65]: Update: i dont see a benefit to replacing the for as a .map

File: app/helpers/page-guard.js (hunk 5 above)
Comment (zavies97) [reply app/helpers/page-guard.js:81]: Update: i dont see a benefit to replacing the for as a .map

File: app/helpers/page-guard.js (hunk 6 above)
Comment (zavies97) [reply app/helpers/page-guard.js:95]: Update: i dont see a benefit to replacing the for as a .map

**Review by yunusfirat — APPROVED** [review]

**Pull request merged**